src/types.ts

```diff
@@ -0,0 +1,103 @@
+export interface SourceFile {
```

Comment:
I put all the type definitions in a `types.ts` file. I'm not sure what the right way™ is in TS, and I couldn't find good examples/conventions here.

Reply:
There is no right™️ way, but this is a common way to do it.